build.assets/tooling/cmd/resource-ref-generator/reference/reference.tmpl

```diff
@@ -0,0 +1,53 @@
+---
+title: {{ .Resource.SectionName }} Reference
+description: Provides a reference of fields within the {{ .Resource.SectionName }} resource, which you can manage with tctl.
+sidebar_title: {{ .Resource.SectionName }}
+---
+{{- if ne .Resource.Kind "" }}
+
+**Kind**: {{ .Resource.Kind }}
+{{- end }}
+
+{{- if ne .Resource.Version "" }}
+**Version**: {{ .Resource.Version }}
+{{- end }}
+
+{{ .Resource.Description }}
+
+{/*Automatically generated from: {{ .Resource.SourcePath}}*/}
```

Comment:
Should we move this line to the top of the file and make it a bit louder so people don't unintentionally try to edit this file?

I'm not sure if gopls operates on .tmpl files, but if it does there's a standard "code generated" comment that will actually tell editors to warn users who try to edit generated files.

Reply:
This is what I was thinking of: https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source

> To convey to humans and machine tools that code is generated, generated source should have a line that matches the following regular expression (in Go syntax):
>
> ```
> ^// Code generated .* DO NOT EDIT\.$
> ```

Reply:
Thanks! The generator adds the "DO NOT EDIT" line to the top of `.mdx` files, so it shouldn't need to match the pattern intended for autogenerated Go files.